### ISSUE TITLE
Add support for Basic Authentication

### ISSUE THREAD

**Issue by captn3m0** [izderadicka/audioserve-android#7]
I Run audioserve with the `--no-authentication` behind a reverse proxy that handles Basic Authentication. This lets me easily use the web interface in most places.

However, the Android application only accepts URL+Secret for connection. I attempted using `https://username:password@audioserveURL` along with an empty secret, but it didn't work.

Happy to share credentials over email if required.

**Comment by izderadicka** [comment]
Yes Basic authentican will not work here, as audioserve is using same header (Authorization) and android client enforces this header.
Best solution would be to make audioserve authentication optional here - if secret is empty in settings, do not add this custom header.  But not sure if HTTP client supports basic authentication in URL - this has to be tested.
